### PR TITLE
[MRG] convert some more % strings to fstrings

### PR DESCRIPTION
**Pull request by sappelhoff**
- follow up to: https://github.com/mne-tools/mne-python/pull/12464
- xref: https://github.com/mne-tools/mne-python/issues/12360

As far as I can see there are around 300 more fixes in around 80 more files to be done. But many of them cannot safely be done automatically, so here are a few manual fixes that should be fine.

### CODE REVIEW

**Review by mscheltienne — APPROVED** [review]

File: mne/_fiff/meas_info.py [hunk 1]
```diff
@@ -1707,7 +1707,7 @@ def __repr__(self):
                 else:
                     entr = v.strftime("%Y-%m-%d %H:%M:%S %Z")
             elif k == "kit_system_id" and v is not None:
-                entr = "%i (%s)" % (v, KIT_SYSNAMES.get(v, "unknown"))
+                entr = f"{v} ({KIT_SYSNAMES.get(v, "unknown")})"
```
Comment (sappelhoff) [inline mne/_fiff/meas_info.py:1710]:
```suggestion
                entr = f"{v} ({KIT_SYSNAMES.get(v, 'unknown')})"
```

File: mne/evoked.py [hunk 2]
```diff
@@ -1900,7 +1900,7 @@ def _read_evoked(fname, condition=None, kind="average", allow_maxshield=False):
         if nepoch != 1 and nepoch != info["nchan"]:
             raise ValueError(
                 "Number of epoch tags is unreasonable "
-                "(nepoch = %d nchan = %d)" % (nepoch, info["nchan"])
+                f"(nepoch = {nepoch} nchan = {info["nchan"]})"
```
Comment (sappelhoff) [inline mne/evoked.py:1903]:
```suggestion
                f"(nepoch = {nepoch} nchan = {info['nchan']})"
```

**Comment by sappelhoff** [comment]
I'll try to add a few more later

**Comment by sappelhoff** [comment]
Still around 150 over 20+ files. 😬 

You can find them by globally searching for `" %` ... ` % ` yields too many modulo operations (not str formatting related)

File: Makefile [hunk 3]
```diff
@@ -4,7 +4,7 @@ PYTHON ?= python
 PYTESTS ?= py.test
 CODESPELL_SKIPS ?= "doc/_build,doc/auto_*,*.fif,*.eve,*.gz,*.tgz,*.zip,*.mat,*.stc,*.label,*.w,*.bz2,*.annot,*.sulc,*.log,*.local-copy,*.orig_avg,*.inflated_avg,*.gii,*.pyc,*.doctree,*.pickle,*.inv,*.png,*.edf,*.touch,*.thickness,*.nofix,*.volume,*.defect_borders,*.mgh,lh.*,rh.*,COR-*,FreeSurferColorLUT.txt,*.examples,.xdebug_mris_calc,bad.segments,BadChannels,*.hist,empty_file,*.orig,*.js,*.map,*.ipynb,searchindex.dat,install_mne_c.rst,plot_*.rst,*.rst.txt,c_EULA.rst*,*.html,gdf_encodes.txt,*.svg,references.bib,*.css,*.edf,*.bdf,*.vhdr"
 CODESPELL_DIRS ?= mne/ doc/ tutorials/ examples/
-all: clean inplace test test-doc
+all: clean test-doc
```
Comment (sappelhoff) [inline Makefile:7]: attention: These are non-f-string-related changes ☝️ 

These targets do not exist in the Makefile and were giving me errors, so I suggest to remove them.

**Comment by sappelhoff** [comment]
@mscheltienne @cbrnr please merge if you are happy with this. I'll fix potential errors that the CI spits out, but I won't be fixing more f-strings ... we'll have to continue with that another time :-)

File: mne/_fiff/_digitization.py [hunk 4]
```diff
@@ -225,7 +225,7 @@ def write_dig(fname, pts, coord_frame=None, *, overwrite=False, verbose=None):
         if len(bad_frames) > 0:
             raise ValueError(
                 "Points have coord_frame entries that are incompatible with "
-                "coord_frame=%i: %s." % (coord_frame, str(tuple(bad_frames)))
+                f"coord_frame={coord_frame}: {str(tuple(bad_frames))}."
```
Comment (larsoner) [inline mne/_fiff/_digitization.py:228]: Some of these I think could be changed to this, but not 100% necessary
```suggestion
                f"coord_frame={coord_frame}: {tuple(bad_frames)}."
```

**Review by larsoner — APPROVED** [review]
I only spot checked but looks good to me!

File: mne/_fiff/_digitization.py (hunk 4 above)
Comment (sappelhoff) [reply mne/_fiff/_digitization.py]: thanks! added in https://github.com/mne-tools/mne-python/pull/12758/commits/ab42a75bbc705795eba670a061b7530728c2007e

**Review by mscheltienne — APPROVED** [review]
Spot check as well, LGTM, marking for merge, thanks @sappelhoff!

**Pull request merged**